clang/lib/Sema/SemaLambda.cpp

```diff
@@ -1888,6 +1888,11 @@ ExprResult Sema::BuildCaptureInit(const Capture &Cap,
 ExprResult Sema::ActOnLambdaExpr(SourceLocation StartLoc, Stmt *Body) {
   LambdaScopeInfo LSI = *cast<LambdaScopeInfo>(FunctionScopes.back());
   ActOnFinishFunctionBody(LSI.CallOperator, Body);
+
+  if (const auto FX = LSI.CallOperator->getFunctionEffects()) {
+    CheckAddCallableWithEffects(LSI.CallOperator, FX);
+  }
```

Comment:
Seems weird to me to do this here seeing as a `LambdaExpr` contains a `CXXRecordDecl` and a call operator, i.e. a `CXXMethodDecl`; shouldn’t this just be done automatically when we create the call operator?

Reply:
will have to revisit this in the follow-on PR